### ISSUE TITLE
Add Tags for Release Versions

### ISSUE THREAD

**Issue by asayler** [trampgeek/moodle-qtype_coderunner#12]
Would it be possible to tag the release versions of this repo in git? This makes it much simpler to clone a specific version of the repo and eases the burden of downstream release management, especially for those of us building Moodle installs via scripts (e.g. for Docker).

E.g.
Running:

```
git tag 2016071501 4761d636cd6ac86daeab1a3736657c0995f87376
git push --tags
```

Allows one to replace:

```
git clone --recursive --single-branch \
"https://github.com/trampgeek/moodle-qtype_coderunner.git" \
"${MOODLE_CODE_PATH}/question/type/coderunner" \
&& cd "${MOODLE_CODE_PATH}/question/type/coderunner" \
&& git checkout 4761d636cd6ac86daeab1a3736657c0995f87376 \
&& cd
```

With the cleaner:

```
git clone --recursive --single-branch -b 2016071501 \
"https://github.com/trampgeek/moodle-qtype_coderunner.git" \
"${MOODLE_CODE_PATH}/question/type/coderunner"
```

I'd file this as a pull quest, but github does not currently allow importing git metadata such as tags via pull request -- tags can only be applied by someone with commit access to the repo itself.

**Comment by trampgeek** [comment]
Hi Andy

Thanks for the suggestion. I chose to set a tag of V3.0.1 as I thought 
that a version number rather than a name might make more sense than a 
date to most users. I included the date, 2016071501, in the tag message. 
However, thinking about it I'm not quite so sure that what I've done 
will fit your needs. If not, please suggest what you'd like me to do. I 
could add 2016071501 as another tag, I guess, but having two different 
tagging streams seems a bit clumsy.

I set a similar tag on the associated question behaviour.

I'm not all that familiar with tags so I'm happy to take suggestions on 
how I might use them better.

Richard

**Comment by asayler** [comment]
"V3.0.1" should be a fine tag. I just need something to feed to git that maps to a fixed commit, and that seems to fit the bill. Thanks!
